lib/gruf/interceptors/instrumentation/request_logging/interceptor.rb

```diff
@@ -49,12 +56,20 @@ def call
               yield
             end
 
-            if result.successful?
-              type = options.fetch(:success_log_level, :info).to_sym
+            # Fetch from options if they have been supplied else default to internally defined mappings...
+            info_level_map = options.fetch(:info_level_codes, LOG_LEVEL_MAPS[:info_level_codes])
+            warn_level_map = options.fetch(:warn_level_codes, LOG_LEVEL_MAPS[:warn_level_codes])
+            error_level_map = options.fetch(:error_level_codes, LOG_LEVEL_MAPS[:error_level_codes])
+
+            type = :error
+            status_name = result.message_class_name
+            if result.successful? || info_level_map.include?(result.message.class)
```

Comment:
Any reason to check `successful?` here since we can just check if the response status is in the info_level_map?

Reply:
I was trying to prevent any code depending on that from failing. The successful flag previously was the only toggle between :error and :info. Now there's :warn. Anyone using this code as before should see it behave the same. That is, any successful result, regardless of specific code will still work as before. Since, however, we are augmenting the version, we can ignore it and make sure to note that in the change log, and just go by the response status. I'll make the correction.